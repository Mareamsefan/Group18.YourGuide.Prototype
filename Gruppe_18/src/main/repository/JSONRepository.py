import json


class JSONRepository:

    def save_to_json(self, entity, filename):
        try:
            with open(filename, "r") as json_file:
                filedata = json.load(json_file)
        except FileNotFoundError:
            filedata = []

        duplicate_found = False
        for item in filedata:
            if item == entity.to_dict():
                duplicate_found = True
                break

        if not duplicate_found:
            data = entity.to_dict()
            filedata.append(data)
        with open(filename, "w") as json_file:
            json.dump(filedata, json_file, indent=4)

    def save_to_stream(self, io_stream):
        try:
            filedata = json.load(io_stream)
        except json.JSONDecodeError:
            filedata = []

        duplicate_found = False
        for item in filedata:
            if item == self.to_dict():
                duplicate_found = True
                break

        if not duplicate_found:
            data = self.to_dict()
            filedata.append(data)

        io_stream.truncate(0)
        io_stream.seek(0)
        json.dump(filedata, io_stream, indent=4)

    def to_dict(self, entity):
        entity_dict = {}
        for key, value in entity.__dict__.items():
            entity_dict[key] = value
        return entity_dict

    def read_from_stream(self, stream):
        items = json.load(stream)
        stream.seek(0)
        return items




    # def from_json(self, entity, ):
