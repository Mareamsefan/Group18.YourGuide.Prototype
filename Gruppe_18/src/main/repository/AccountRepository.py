from Gruppe_18.src.main.repository.JSONRepository import JSONRepository
from Gruppe_18.src.main.model.models import Account


class AccountRepository (JSONRepository):
    def delete_account(self, session, entity):
        account_to_delete = session.query(Account).filter_by(username=entity.username).first()

        if account_to_delete:
            session.delete(account_to_delete)
            session.commit()
            return True

        return False

    '''

    def delete_account(self, entity, io_stream):
        try:
                io_stream.seek(0)
                filedata = json.load(io_stream)
        except json.JSONDecodeError:
            filedata = []

        index_to_delete = None
        for i, item in enumerate(filedata):
            if item["username"] == entity.username:
                index_to_delete = i
                break

        if index_to_delete is not None:
            del filedata[index_to_delete]

            io_stream.seek(0)
            io_stream.truncate()
            json.dump(filedata, io_stream, indent=4)
            return True

        return False
'''
    # needs to be updated after database is implemented

    def successful_registration(self, entity, io_stream):
        self.save_to_stream(entity, io_stream)
        return True

