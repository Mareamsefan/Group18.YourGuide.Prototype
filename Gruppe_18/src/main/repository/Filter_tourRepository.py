from Gruppe_18.src.main.repository.JSONRepository import JSONRepository

class FilterRepository (JSONRepository):
    def read_from_stream(self, stream):
