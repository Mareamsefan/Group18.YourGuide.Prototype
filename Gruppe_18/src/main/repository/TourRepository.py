from Gruppe_18.src.main.repository.JSONRepository import JSONRepository

list_with_tours =

class TourRepository (JSONRepository):

    def filter_tour_by_location(tour_list, destination):
        filtered_tours = []
        for tour in tour_list:
            if tour['destination'] == destination:
                filtered_tours.append(tour)
        return filtered_tours

    def filter_tour_by_price(tour_list, max_price, min_price):
        filtered_tours = []
        for tour in tour_list:
            if min_price <= tour['cost'] <= max_price:
                filtered_tours.append(tour)
        return filtered_tours