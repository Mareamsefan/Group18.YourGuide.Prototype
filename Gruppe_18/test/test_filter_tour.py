from Gruppe_18.test.stream_for_testing import *
from approvaltests import verify
from Gruppe_18.src.main.repository.TourRepository import TourRepository



def test_if_reading_from_stream_is_as_expected():
    tours = TourRepository.read_tours_from_stream(stream)
    verify(tours)


def test_if_filter_tours_by_location_is_as_expected():
    filtered = TourRepository.filter_tour_by_location(read_tours_from_stream(stream), "London")
    verify(filtered)


def test_if_filter_tours_by_price_is_as_expected():
    filtered = TourRepository.filter_tour_by_price(read_tours_from_stream(stream), 150, 0)
    verify(filtered)
