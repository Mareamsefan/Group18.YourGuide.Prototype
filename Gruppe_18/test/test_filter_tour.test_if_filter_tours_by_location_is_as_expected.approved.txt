[{'destination': 'London', 'duration': 3, 'cost': 180, 'pictureURL': 'https://example.com/london.jpg', 'language': 'English', 'max_travelers': 18, 'booked': 6, 'tour_Id': 4}, {'destination': 'London', 'duration': 3, 'cost': 130, 'pictureURL': 'https://example.com/london.jpg', 'language': 'English', 'max_travelers': 19, 'booked': 4, 'tour_Id': 5}]
